specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-03-01-preview/costallocation.json

```diff
@@ -0,0 +1,689 @@
+        "Invalid",
+        "AlreadyExists",
+        "Valid"
+      ],
+      "x-ms-enum": {
+        "name": "Reason",
+        "modelAsString": true
+      }
+    },
+    "CostAllocationRuleDefinition": {
+      "type": "object",
+      "description": "The cost allocation rule model definition",
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure Resource Manager Id for the rule. This is a read ony value."
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Name of the rule. This is a read only value."
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Resource type of the rule. This is a read only value of Microsoft.CostManagement/CostAllocationRule."
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "CostAllocationRuleDefinitionResponse": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/CostAllocationRuleDefinition"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "properties": {
+              "$ref": "#/definitions/CostAllocationRulePropertiesResponse",
+              "description": "Cost allocation rule properties"
+            }
+          }
+        }
+      ],
+      "description": "The cost allocation rule model definition"
+    },
+    "CostAllocationRuleDefinitionRequest": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/CostAllocationRuleDefinition"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "properties": {
+              "$ref": "#/definitions/CostAllocationRulePropertiesRequest",
+              "description": "Cost allocation rule properties"
+            }
+          }
+        }
+      ],
+      "description": "The cost allocation rule model definition"
+    },
+    "CostAllocationRuleProperties": {
+      "type": "object",
+      "description": "The properties of a cost allocation rule",
+      "additionalProperties": false,
+      "required": [
+        "details"
+      ],
+      "properties": {
+        "description": {
+          "description": "Description of a cost allocation rule.",
+          "type": "string"
+        },
+        "details": {
+          "description": "Resource information for the cost allocation rule",
+          "$ref": "#/definitions/CostAllocationRuleDetails"
+        }
+      }
+    },
+    "CostAllocationRulePropertiesResponse": {
+      "description": "The properties of a cost allocation rule response",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CostAllocationRuleProperties"
+        },
+        {
+          "type": "object",
+          "properties": {
+            "status": {
+              "$ref": "#/definitions/RuleStatus",
+              "description": "Status of the rule"
+            },
+            "createdDate": {
```

Comment:
there is a new top level property called systemData. and such fields are part of the schema there. This is dont for consistency sake so that any azure resource will have this data and in the same property. https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v2/types.json#L355
Other details here - https://armwiki.azurewebsites.net/api_contracts/ResourceSystemData.html?q=systemdata

Reply:
Is this change required? We already reviewed having the field this way previously (https://github.com/Azure/azure-rest-api-specs-pr/pull/1152) and now have a number of dependencies based on this. Since we determined our setup for this before the change, it introduces lots of randomization on us to switch post development